hibernate-core/src/test/java/org/hibernate/orm/test/entitygraph/FetchGenericAttributesTest.java

```diff
@@ -0,0 +1,253 @@
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.MappedSuperclass;
+import jakarta.persistence.OneToMany;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hibernate.testing.hamcrest.InitializationCheckMatcher.isInitialized;
+
+@DomainModel(
+		annotatedClasses = {
+				FetchGenericAttributesTest.AbsOne.class,
+				FetchGenericAttributesTest.AbsTwo.class,
+				FetchGenericAttributesTest.One.class,
+				FetchGenericAttributesTest.Two.class,
+				FetchGenericAttributesTest.Three.class
+		}
+)
+@SessionFactory
+@JiraKey("HHH-18151")
+public class FetchGenericAttributesTest {
+
+	private Long oneId;
+	private Long threeId;
+
+	@BeforeEach
+	void before(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					final var one = new One();
+					final var two = new Two();
+					final var three = new Three();
+					one.setTwo( two );
+					two.getOnes().add( one );
+					two.setThree( three );
+					three.getTwos().add( two );
+					one.setOneProp( "oneProp" );
+					two.setTwoProp( "twoProp" );
+					three.setThreeProp( "threeProp" );
+
+					session.persist( one );
+					session.persist( two );
+					session.persist( three );
+					oneId = one.getId();
+					threeId = three.getId();
+				}
+		);
+	}
+
+	@AfterEach
+	void after(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					session.createMutationQuery( session.getCriteriaBuilder().createCriteriaDelete( One.class ) )
+							.executeUpdate();
+					session.createMutationQuery( session.getCriteriaBuilder().createCriteriaDelete( Two.class ) )
+							.executeUpdate();
+					session.createMutationQuery( session.getCriteriaBuilder().createCriteriaDelete( Three.class ) )
+							.executeUpdate();
+				}
+		);
+	}
+
+	@Test
+	@JiraKey("HHH-18151")
+	void HHH18151WithGraphParser(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					One one = session.byId( One.class )
+							.with( GraphParser.parse( One.class, "two(three)", session ), GraphSemantic.FETCH )
+							.load( oneId );
+
+					assertThat( one.getId(), is( oneId ) );
+					assertThat( one.getTwo(), isInitialized() );
+					assertThat( one.getTwo().getThree(), isInitialized() );
+				}
+		);
+	}
+
+	@Test
+	@JiraKey("HHH-18151")
+	void HHH18151WithEntityGraph(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					final var oneGraph = session.createEntityGraph( One.class );
+
+					//before the fix you had to use method
```

Comment:
@mbladel should I remove this comment?

Reply:
Yeah, the `addSubgraph(String, Class)` method is part of the JPA `EntityGraph`  interface and is useful when dealing with inheritance types so we won't be deprecating it.